libraries/AP_Scripting/lua_bindings.cpp

```diff
@@ -532,3 +535,65 @@ int lua_removefile(lua_State *L) {
   const char *filename = luaL_checkstring(L, 1);
   return luaL_fileresult(L, remove(filename) == 0, filename);
 }
+
+
+// local cmd, data = crsf_pop() -- get byte array from crsf or nil if no data received
+int lua_crsf_pop(lua_State *L) {
+#if HAL_CRSF_TELEM_ENABLED
+    AP_RCProtocol_CRSF *crsf = AP_RCProtocol_CRSF::get_singleton();
+
+    // Allocate buffer on first call to this function
+    if (!crsf->lua_pop_buffer) {
+        crsf->lua_pop_buffer = new AP_RCProtocol_CRSF::lua_buffer_t();
+    }
+
+    // Move buffer data to lua
+    AP_RCProtocol_CRSF::lua_buffer_t *buf = crsf->lua_pop_buffer;
+    if (buf && buf->len > 0) {
+        WITH_SEMAPHORE(buf->sem);
+        lua_pushinteger(L, buf->cmd);
+        lua_pushlstring(L, buf->data, buf->len);
+        buf->len = 0;
+    } else {
+        lua_pushnil(L);
+        lua_pushnil(L);
+    }
+#else
+    lua_pushnil(L);
+    lua_pushnil(L);
+#endif
+    return 2;
+}
+
+// local num_bytes = crsfPush(cmd, data) -- push byte array to crsf
+int lua_crsf_push(lua_State *L) {
+#if HAL_CRSF_TELEM_ENABLED
+    int push_len = 0;
+    int isnum;
+    uint8_t cmd = lua_tointegerx(L, 1, &isnum);
```

Comment:
you can use the helper `get_uint8_t(L, 1)` It will not return if value is invalid.